sktime/forecasting/compose/_enbpi_forecaster.py

```diff
@@ -0,0 +1,219 @@
+
+
+    References
+    ----------
+    .. [1] Chen Xu & Yao Xie (2021). Conformal Prediction Interval for Dynamic
+    Time-Series.
+    .. [2] Valeriy Manokhin, PhD, MBA, CQF. Demystifying EnbPI: Mastering Conformal
+    Prediction Forecasting
+    """
+
+    _tags = {
+        "authors": ["benheid"],
+        "python_dependencies": ["tsbootstrap>=0.1.0", "fortuna", "jax"],
+        "scitype:y": "univariate",  # which y are fine? univariate/multivariate/both
+        "ignores-exogeneous-X": False,  # does estimator ignore the exogeneous X?
+        "handles-missing-data": False,  # can estimator handle missing data?
+        "y_inner_mtype": "pd.DataFrame",
+        # which types do _fit, _predict, assume for y?
+        "X_inner_mtype": "pd.DataFrame",
+        # which types do _fit, _predict, assume for X?
+        "X-y-must-have-same-index": True,  # can estimator handle different X/y index?
+        "requires-fh-in-fit": True,  # like AutoETS overwritten if forecaster not None
+        "enforce_index_type": None,  # like AutoETS overwritten if forecaster not None
+        "capability:insample": False,  # can the estimator make in-sample predictions?
+        "capability:pred_int": True,  # can the estimator produce prediction intervals?
+        "capability:pred_int:insample": False,  # ... for in-sample horizons?
+    }
+
+    def __init__(self, forecaster=None, bootstrap_transformer=None, random_state=None):
+        self.forecaster = forecaster
+        self.forecaster_ = (
+            clone(forecaster) if forecaster is not None else NaiveForecaster()
+        )
+        self.bootstrap_transformer = bootstrap_transformer
+        self.random_state = random_state
+
+        super().__init__()
+
+        from tsbootstrap import MovingBlockBootstrap
+
+        self.bootstrap_transformer_ = (
+            clone(bootstrap_transformer)
+            if bootstrap_transformer is not None
+            else MovingBlockBootstrap()
+        )
+
+    def _fit(self, X, y, fh=None):
+        self._fh = fh
+        self._y_ix_names = y.index.names
+
+        # random state handling passed into input estimators
+        self.random_state_ = check_random_state(self.random_state)
+
+        # fit/transform the transformer to obtain bootstrap samples
+        bs_ts_index = list(
+            self.bootstrap_transformer_.bootstrap(y, test_ratio=0, return_indices=True)
+        )
+        self.indexes = np.stack(list(map(lambda x: x[1], bs_ts_index)))
+        bootstapped_ts = list(map(lambda x: x[0], bs_ts_index))
+
+        self.forecasters = []
+        self.preds = []
+        # Fit Models per Bootstrap Sample
+        for bs_ts in bootstapped_ts:
+            bs_df = pd.DataFrame(bs_ts, index=y.index)
+            forecaster = clone(self.forecaster_)
+            forecaster.fit(y=bs_df, fh=fh, X=X)
+            self.forecasters.append(forecaster)
+            prediction = forecaster.predict(fh=y.index, X=X)
```

Comment:
why are we making predictions here? It does not seem the `self.preds` is ever used again.

Reply:
Done